third_party/move/move-compiler-v2/tests/checking/inlining/function_name_shadowing.move

```diff
@@ -0,0 +1,17 @@
+//# publish
+module 0x42::Test {
+
+    public fun f(a: u64, b: u64): u64 {
+        a * b
+    }
+
+    public inline fun quux(f:|u64, u64|u64, a: u64, b: u64): u64 {
+        f(a, b)
```

Comment:
This is rather tangential to this PR, but I've added use statements.  The explicit qualification case is not interesting, as we know what it will do.

What is happening is very clear if you look at `move-compiler/src/expander/translation.rs` and `.../aliases.rs`, which tracks nesting of `use` of symbols from other modules.  All that is tracked are those imported symbols, not any local variables.  A name in function call position (and a few other places) is turned into an explicit `ModuleIdent::Name` pair.  A name in other positions may be left unevaluated until later.